src/PayID/pay-id-client.ts

```diff
@@ -24,28 +22,36 @@ interface PayIDComponents {
  *
  * @warning This class is experimental and should not be used in production applications.
  */
-export default class PayIDClient implements PayIDClientInterface {
+export default class PayIDClient {
   /**
+   * Initialize a new PayID client.
+   *
+   * Networks in this constructor take the form of an asset and an optional network (<asset>-<network>), for instance:
+   * - xrpl-testnet
+   * - xrpl-mainnet
+   * - eth-rinkby
+   * - ach
+   *
+   * TODO(keefertaylor): Link a canonical list at payid.org when available.
+   *
    * @param network The network that addresses will be resolved on.
    */
-  constructor(public readonly network: XRPLNetwork) {}
+  constructor(public readonly network: string) {}
 
   /**
-   * Retrieve the XRP Address associated with a PayID.
-   *
-   * @note The returned value will always be in an X-Address format.
+   * Retrieve the address associated with a PayID.
    *
    * @param payID The payID to resolve for an address.
-   * @returns An XRP address representing the given PayID.
+   * @returns An address representing the given PayID.
    */
-  async xrpAddressForPayID(payID: string): Promise<string> {
+  async addressForPayID(payID: string): Promise<string> {
     const payIDComponents = PayIDClient.parsePayID(payID)
 
     const client = new ApiClient()
     client.basePath = `https://${payIDComponents.host}`
 
     // Accept only the given network in response.
-    const accepts = [`application/xrpl-${this.network}+json`]
+    const accepts = [`application/${this.network}+json`]
```

Comment:
So `this.network` = `xrpl-mainnet` or `ach`?

Reply:
correct. We leave it as a generic string so that you can place whatever you'd like into this.